includes/abstracts/abstract-wc-product.php

```diff
@@ -27,7 +27,7 @@ class WC_Product extends WC_Abstract_Legacy_Product {
 	 * @var array
 	 */
 	protected $data = array(
-		'name'               => '',
+		'name'               => 'New Product',
```

Comment:
I found out that without a title/content the post will fail to create on save. It's seemingly required by WordPress, so I just added a placeholder here.